src/Gui/Inventor/Draggers/SoRotationDraggerGeometry.cpp

```diff
@@ -0,0 +1,380 @@
+}
+
+SoRotatorGeometry::SoRotatorGeometry()
+{
+    SO_KIT_CONSTRUCTOR(SoRotatorGeometry);
+    SO_KIT_ADD_CATALOG_ENTRY(separator, SoSeparator, false, this, "", false);
+    SO_KIT_ADD_CATALOG_ENTRY(lightModel, SoLightModel, false, separator, "", false);
+    SO_KIT_ADD_CATALOG_ENTRY(pickStyle, SoPickStyle, false, separator, "", false);
+    SO_KIT_ADD_CATALOG_ENTRY(drawStyle, SoDrawStyle, false, separator, "", false);
+    SO_KIT_ADD_CATALOG_ENTRY(arcCoords, SoCoordinate3, false, separator, "", true);
+    SO_KIT_ADD_CATALOG_ENTRY(arc, SoLineSet, false, separator, "", true);
+    SO_KIT_ADD_CATALOG_ENTRY(pivotSeparator, SoSeparator, false, separator, "", true);
+    SO_KIT_ADD_CATALOG_ENTRY(rotorPivot, SoSphere, false, pivotSeparator, "", true);
+
+    SO_KIT_ADD_CATALOG_ENTRY(_rotorPivotTranslation, SoTranslation, false, pivotSeparator, rotorPivot, false);
+
+    SO_KIT_ADD_FIELD(arcRadius, (8.0f));
+    SO_KIT_ADD_FIELD(arcAngle, (std::numbers::pi_v<float> / 2.0f));
+    SO_KIT_ADD_FIELD(sphereRadius, (0.8f));
+    SO_KIT_ADD_FIELD(arcThickness, (4.0f));
+
+    SO_KIT_INIT_INSTANCE();
+
+    auto lightModel = SO_GET_ANY_PART(this, "lightModel", SoLightModel);
+    lightModel->model = SoLightModel::BASE_COLOR;
+
+    auto pickStyle = SO_GET_ANY_PART(this, "pickStyle", SoPickStyle);
+    pickStyle->style = SoPickStyle::SHAPE_ON_TOP;
+
+    auto rotorPivot = SO_GET_ANY_PART(this, "rotorPivot", SoSphere);
+    rotorPivot->radius.connectFrom(&sphereRadius);
+
+    auto drawStyle = SO_GET_ANY_PART(this, "drawStyle", SoDrawStyle);
+    drawStyle->lineWidth.connectFrom(&arcThickness);
+
+    auto translation = SO_GET_ANY_PART(this, "_rotorPivotTranslation", SoTranslation);
+    pivotPosition.connectFrom(&translation->translation);
+
+    auto arc = SO_GET_ANY_PART(this, "arc", SoLineSet);
+    arc->numVertices = segments + 1;
+
+    // forces the notify method to get called so that the initial translations and other values are set
+    arcRadius.touch();
+}
+
+void SoRotatorGeometry::notify(SoNotList* notList)
+{
+    assert(notList);
+    SoField* lastField = notList->getLastField();
+
+    if (lastField == &arcRadius || lastField == &arcAngle) {
+        float angle = arcAngle.getValue();
+        float radius = arcRadius.getValue();
+
+        auto coordinates = SO_GET_ANY_PART(this, "arcCoords", SoCoordinate3);
+        float angleIncrement = angle / static_cast<float>(segments);
+        SbRotation rotation(SbVec3f(0.0, 0.0, 1.0), angleIncrement);
+        SbVec3f point(radius, 0.0, 0.0);
+        for (int index = 0; index <= segments; ++index) {
+            coordinates->point.set1Value(index, point);
+            rotation.multVec(point, point);
+        }
+
+        auto translation = SO_GET_ANY_PART(this, "_rotorPivotTranslation", SoTranslation);
+        translation->translation = {radius * cos(angle / 2.0f), radius * sin(angle / 2.0f), 0};
+    }
+}
+
+SO_KIT_SOURCE(SoRotatorGeometry2)
+
+void SoRotatorGeometry2::initClass()
```

Comment:
What does `2` mean? If it is TwoSided spell it out, using numbers in names is bad practice.